src/layers/graph-layer/graph-layer.js

```diff
@@ -0,0 +1,303 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {BrushingExtension} from '@deck.gl/extensions';
+import {IconLayer, LineLayer} from '@deck.gl/layers';
+
+import Layer from '../base-layer';
+import {hexToRgb} from 'utils/color-utils';
+import {findDefaultColorField} from 'utils/dataset-utils';
+import GraphLayerIcon from './graph-layer-icon';
+import {DEFAULT_LAYER_COLOR} from 'constants/default-settings';
+
+export const iconPosAccessor = ({icon}) => dc => d => {
+  if (!icon) {
+    return null;
+  }
+  return {
+    url: dc.valueAt(d.index, icon.fieldIdx),
+    width: 64,
+    height: 64,
+    anchorY: 64
```

Comment:
Image used in the example does not match with this resolution. Makes it squashed.
![image](https://user-images.githubusercontent.com/3582318/183861382-f9601cd5-3d67-4dab-99d1-c00b401c01aa.png)

As this will highly depend on the icon being used this. We can keep it hardcoded for now, but we will have to look into using a precreated icon/texture atlas or something that can do this dynamically

Reply:
Per node the icon is defined, in this case it's all the same

Reply:
@ismakutl Makes a lot of sense, but will the dimensions also be part of the node definition? As deckgl needs to know what size and aspect ratio to draw it with.

Maybe the node metadata should only specify a node type. Which then maps to an icon url with it's dimensions in a lookup file.